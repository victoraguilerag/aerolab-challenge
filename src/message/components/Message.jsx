import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

class Message extends Component {
  render () {
    const { message } = this.props
    return (
      <div className="message" style={styles.message}>
        { message }
      </div>
    )
  }
}

const styles = {
  message: {
    'position': 'fixed',
    'width': '250px',
    'height': '50px',
    'right': '25px',
    'bottom': '10px',
    'background': 'white',
    'color': 'rgb(10, 212, 250)',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'borderRadius': '50px',
    'boxShadow': '2px 0px 5px rgba(150,150,150,.5)',
    'zIndex': '200'
  }
}

Message.propTypes = {
  message: PropTypes.string.isRequired
}

const mapStateToProps = (state) => {
  return {
    message: state.message
  }
}

export default connect(mapStateToProps)(Message)
