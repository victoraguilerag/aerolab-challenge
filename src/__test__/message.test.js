import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import Message from '../message/components/Message.jsx'
import productsStub from '../stub/products.js'

const mockStore = configureStore()
const initialState = {
  message: 'Message displayed'
}

const store = mockStore(initialState)

test('Message renders', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Message  />
    </Provider>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
