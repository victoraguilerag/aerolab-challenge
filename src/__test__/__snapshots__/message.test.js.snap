// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders 1`] = `
<div
  className="message"
  style={
    Object {
      "alignItems": "center",
      "background": "white",
      "borderRadius": "50px",
      "bottom": "10px",
      "boxShadow": "2px 0px 5px rgba(150,150,150,.5)",
      "color": "rgb(10, 212, 250)",
      "display": "flex",
      "height": "50px",
      "justifyContent": "center",
      "position": "fixed",
      "right": "25px",
      "width": "250px",
      "zIndex": "200",
    }
  }
>
  Message displayed
</div>
`;
